docs/npm-workspaces.md

```diff
@@ -0,0 +1,72 @@
+- `--if-present`: you will notice this on a few of the scripts added in the root [package.json](./package.json). This is because when deploying individual packages to a Docker container, only the package and any shared [libs](./libs) directories will be available from the full list of workspaces in the root [package.json](./package.json)
+
+## Notes on node_modules hoisting limitations
+
+One of the benefits of `npm workspaces` is that common dependencies across all sub-packages will be hoisted to the root `node_modules` with only conflicting dependencies installed in the sub-package `node_modules`. This does however present some challenges. Below are some notes on challenges faced and how we've mitigated them:
+
+Our UI packages (`portal`, `gef-ui` and `trade-finance-manager-ui`) make use of the [govuk-frontend](https://www.npmjs.com/package/govuk-frontend) and [@ministryofjustice/frontend](https://www.npmjs.com/package/@ministryofjustice/frontend) libraries for UI components and styles. When importing styles from these packages into our own Sass `.scss` files the docs advise us to import as follows:
+
+```sass
+@import "node_modules/govuk-frontend/govuk/all";
+```
+
+However, the `govuk-frontend` package has now been hoisted to the root `node_modules`, so this path is no longer correct. One way we may think to resolve this is to simply change the import path to:
+
+```sass
+@import "../node_modules/govuk-frontend/govuk/all";
+```
+
+However, this doesn't completely resolve the issue. The file we now import, itself imports other files pointing to `"node_modules/govuk-frontend/govuk/<some_other_sub_directory>"` and we have no real control over changing them.
+
+The solution then was instead to add an alias to our webpack configuration. e.g.
+
+```javascript
+resolve: {
+  alias: {
+    'node_modules/govuk-frontend':path.resolve(__dirname, '../node_modules/govuk-frontend'),
+    'node_modules/@ministryofjustice':path.resolve(__dirname, '../node_modules/@ministryofjustice'),
+  },
+},
+```
+
+so that all files, regardless if in our control or not, will be updated when the minified CSS files are created. We can now continue to use `@import "node_modules/govuk-frontend/govuk/all";` as suggested in the library docs.
+
+Another way this could have been resolved would be to prevent hoisting of these two packages. Unfortunately though in npm workspaces the choice is to hoist all or nothing (see [install-strategy](https://docs.npmjs.com/cli/v9/commands/npm-install#install-strategy)), however in some workspaces managers there are options to prevent hoisting at a package-by-package level (e.g. see the [nohoist](https://classic.yarnpkg.com/blog/2018/02/15/nohoist/) option provided by Yarn Workspaces). Hopefully this functionality will be implemented in npm workspaces in future.
```

Comment:
Maybe we _should_ use `"install-strategy": "nested"` to prevent hoisting of common node_modules to the root and prevent issues like the above.

We'd still get the benefits of simplified commands that run on all workspace projects, and auto-symlinking to enabled shared code.

Reply:
After discussing a few things with you in person I think keeping like this for now is good, my biggest thing both here and in general on DTFS would be some tree shaking of modules and we should re-visit this if we ever do the tree shaking